src/main/java/com/javaworm/configme/FetchedDataHandler.java

```diff
@@ -8,23 +8,38 @@
 import io.fabric8.kubernetes.api.model.OwnerReference;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import io.fabric8.kubernetes.client.dsl.Resource;
+import io.micrometer.core.instrument.MeterRegistry;
+
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.time.LocalDateTime;
 import java.time.ZoneId;
 import java.time.format.DateTimeFormatter;
 import java.util.Base64;
 import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Optional;
 import java.util.zip.GZIPOutputStream;
 
 public class FetchedDataHandler {
   private final KubernetesClient k8sClient;
   private final K8sEventEmitter k8sEventEmitter;
+  private final MeterRegistry registry;
+  private final Map<String, Long> lastUpdateMap = new HashMap<>();
 
-  public FetchedDataHandler(KubernetesClient k8sClient) {
+  public FetchedDataHandler(KubernetesClient k8sClient, MeterRegistry registry) {
     this.k8sClient = k8sClient;
     this.k8sEventEmitter = new K8sEventEmitter(k8sClient);
+    this.registry = registry;
+  }
+
+  private void onLastUpdateChange(String configName, Long lastUpdateValue) {
+    var metricName = "something." + configName;
```

Comment:
If instead of MeterRegistry we inject a service here, that service can take care of the **prefix**